parser_library/src/processing/preprocessors/preprocessor_utils.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2022 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+#include "preprocessor_utils.h"
+
+#include <algorithm>
+
+#include "lexing/logical_line.h"
+#include "semantics/range_provider.h"
+#include "semantics/statement.h"
+#include "utils/string_operations.h"
+
+namespace hlasm_plugin::parser_library::processing {
+namespace {
+size_t get_quoted_string_end(std::string_view s)
+{
+    auto closing_quote = std::string_view::npos;
+
+    s.remove_prefix(1);
+    while (closing_quote == std::string_view::npos)
+    {
+        closing_quote = s.find_first_of("'");
+
+        if (closing_quote == std::string_view::npos || closing_quote == s.length()
```

Comment:
```suggestion
        if (closing_quote == std::string_view::npos || closing_quote + 1 == s.length()
```
find cannot ever return `s.length()`.

Reply:
Thanks!